fast/stages/1-vpc-sc/variables.tf

```diff
@@ -0,0 +1,174 @@
+  type        = number
+  default     = null
+}
+
+variable "egress_policies" {
+  description = "Egress policy definitions that can be referenced in perimeters."
+  type = map(object({
+    from = object({
+      identity_type = optional(string)
+      identities    = optional(list(string))
+    })
+    to = object({
+      operations = optional(list(object({
+        method_selectors     = optional(list(string))
+        permission_selectors = optional(list(string))
+        service_name         = string
+      })), [])
+      resources              = optional(list(string))
+      resource_type_external = optional(bool, false)
+    })
+  }))
+  default  = {}
+  nullable = false
+  validation {
+    condition = alltrue([
+      for k, v in var.egress_policies :
+      v.from.identity_type == null ? true : contains([
+        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
+        "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT", ""
+      ], v.from.identity_type)
+    ])
+    error_message = "Invalid `from.identity_type` value in egress policy."
+  }
+}
+
+variable "factories_config" {
+  description = "Paths to folders that enable factory functionality."
+  type = object({
+    access_levels       = optional(string, "data/access-levels")
+    egress_policies     = optional(string, "data/egress-policies")
+    ingress_policies    = optional(string, "data/ingress-policies")
+    restricted_services = optional(string, "data/restricted-services.yaml")
+  })
+  nullable = false
+  default  = {}
+}
+
+variable "ingress_policies" {
+  description = "Ingress policy definitions that can be referenced in perimeters."
+  type = map(object({
+    from = object({
+      access_levels = optional(list(string), [])
+      identity_type = optional(string)
+      identities    = optional(list(string))
+      resources     = optional(list(string), [])
+    })
+    to = object({
+      operations = optional(list(object({
+        method_selectors     = optional(list(string))
+        permission_selectors = optional(list(string))
+        service_name         = string
+      })), [])
+      resources = optional(list(string))
+    })
+  }))
+  default  = {}
+  nullable = false
+  validation {
+    condition = alltrue([
+      for k, v in var.ingress_policies :
+      v.from.identity_type == null ? true : contains([
+        "IDENTITY_TYPE_UNSPECIFIED", "ANY_IDENTITY",
+        "ANY_USER_ACCOUNT", "ANY_SERVICE_ACCOUNT", ""
+      ], v.from.identity_type)
+    ])
+    error_message = "Invalid `from.identity_type` value in ingress policy."
+  }
+}
+
+variable "outputs_location" {
+  description = "Path where providers, tfvars files, and lists for the following stages are written. Leave empty to disable."
+  type        = string
+  default     = null
+}
+
+variable "perimeters" {
+  description = "Perimeter definitions."
+  type = object({
+    default = optional(object({
+      access_levels    = optional(list(string), [])
+      dry_run          = optional(bool, false)
```

Comment:
This, together with the default access level, can lock you out if you don't update the default regions.

Reply:
set default to `dry_run = true`